guides/Advanced SettingGateway/UnderstandingSchemaFolders.md

```diff
@@ -0,0 +1,164 @@
+
+```javascript
+this.client.gateways.gatewayName.schema.removeKey(name, force);
+```
+
+In case you have a key you do not longer use and you want to get rid of it, for example, the recently created **userBlacklist** key for guild configs, you would run the following code:
+
+```javascript
+this.client.gateways.guilds.schema.removeKey('userBlacklist');
+```
+
+And the property `userBlacklist` for all guild configs will be deleted, that being said:
+
+```javascript
+msg.guild.configs.userBlacklist;
+// undefined
+'userBlacklist' in msg.guild.configs;
+// false
+```
+
+## Adding folders
+
+Folder creation is very similar to key creation, but with one key difference: it has no options for itself, but instead, it can create its children keys (like you can add a box with another boxes and pieces, into another). You can add a new key inside a new folder in two different ways:
+
+### Slower
+
+You can create a folder, then create the keys, however, this will iterate over all entries twice:
+
+```javascript
+async function init() {
+	const { schema } = this.client.gateways.guilds;
+
+	await schema.addFolder('channels');
+	await schema.channels.addKey('modlogs', { type: 'TextChannel' });
+	console.log(schema.channels.modlogs.toJSON());
+	// {
+	//  	type: 'textchannel',
+	//  	array: false,
+	//  	default: null,
+	//  	min: null,
+	//  	max: null,
+	//  	configurable: true
+	// }
+}
+```
+
+### Faster
+
+However, it's possible to create a folder with all the sub-keys (and even more nested folders) with the folder creation.
+
+```javascript
+async function init() {
+	const { schema } = this.client.gateways.guilds;
+
+	await schema.addFolder('channels', { modlogs: { type: 'TextChannel' } });
+	console.log(schema.channels.modlogs.toJSON());
+	// {
+	//  	type: 'textchannel',
+	//  	array: false,
+	//  	default: null,
+	//  	min: null,
+	//  	max: null,
+	//  	configurable: true
+	// }
+}
+```
+
+> **Reminder**: To access to a key inside a folder in your configuration command, you use the access operator (`.`). For example: *k!conf set channels.modlogs #modlogs*
+
+## Removing folders
+
+It's exactly the same as {@link SchemaFolder#removeKey}, but using {@link SchemaFolder#removeFolder} instead. With the following syntax:
+
+```javascript
+this.client.gateways.gatewayName.schema.removeFolder(name, force);
+```
+
+To remove a folder, like the aforementioned **channels** folder, you would run the following code:
+
+```javascript
+this.client.gateways.guilds.schema.removeFolder('channels');
+```
+
+This will remove all the data from all the sub-keys and sub-folders, even very nested ones.
+
+## Ensuring the existence of a key.
+
+In [Klasa-Pieces](https://github.com/dirigeants/klasa-pieces/), specially, some pieces require a key from the configuration to work, however, the creator of the pieces does not know if the user who downloads the piece has it, so this function becomes useful in this case.
```

Comment:
In [Klasa-Pieces](https://github.com/dirigeants/klasa-pieces/) [REMOVE THE COMMA] ***especially***,